akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -752,7 +755,12 @@ abstract class ShardCoordinator(
         sender() ! reply
 
       case ShardCoordinator.Internal.Terminate =>
-        log.debug("Received termination message")
+        if (rebalanceInProgress.isEmpty)
+          log.debug("Received termination message.")
+        else
+          log.debug(
+            "Received termination message. Rebalance in progress of shards [{}].",
+            rebalanceInProgress.keySet.mkString(", "))
```

Comment:
Unrelated observation. I will create separate issue for that. I just noticed that when leaving the coordinator node the regions initiate a graceful shutdown, which trigger rebalance. The coordinator is stopped before that is completed. Works fine anyway, but not optimal (performance).

Reply:
created issue https://github.com/akka/akka/issues/28917